test/compositionGuards.spec.ts

```diff
@@ -0,0 +1,16 @@
+import {assert} from 'chai';
+
+import {isDateOrStringyDate} from '../src';
+
+describe('compositionGuards', function() {
+	let date: Date;
+	beforeEach(function() {
+		date = new Date();
+	});
+	it('should return true for a date', function() {
+		assert.isTrue(isDateOrStringyDate(date));
+	});
+	it('should return true for a stringyDate date', function() {
+		assert.isTrue(isDateOrStringyDate(date.toISOString()));
+	});
```

Comment:
negative cases?  what about an int?  what about a string that's not a date?

Reply:
agree, should have test cases for undefined, null, symbol, array, and object in addition to numbers